varinfo/generate_umm_var.py

```diff
@@ -0,0 +1,92 @@
+''' This module generates UMM-Var records for a specified collection, and
+    optionally will publish those records to CMR.
+    `generation_collection_umm_var` will:
+
+    * Performing a CMR granule search with collection-based query parameters.
+    * Downloading the granule locally.
+    * Parse the variable metadata from the granule using VarInfoFromNetCDF4.
+    * Generate UMM-Var JSON as a dictionary for each identified variable.
+    * (Optional) Publish each UMM-Var entry to the selected CMR environment.
+
+'''
+from tempfile import TemporaryDirectory
+from typing import Dict, List, Union
+import re
+
+from cmr import CMR_UAT
+
+from varinfo import VarInfoFromNetCDF4
+from varinfo.cmr_search import (CmrEnvType, download_granule, get_granule_link,
+                                get_granules)
+from varinfo.umm_var import get_all_umm_var, publish_all_umm_var
+
+
+# Custom return type: either a list of UMM-Var JSON (a list of dictionaries),
+# or a list of strings (either concept IDs or error strings).
+UmmVarReturnType = List[Union[Dict, str]]
+
+
+def generate_collection_umm_var(collection_concept_id: str,
+                                auth_header: str,
+                                cmr_env: CmrEnvType = CMR_UAT,
+                                publish: bool = False) -> UmmVarReturnType:
+    """ Run all the of the functions for downloading and publishing
+        a UMM-Var entry to CMR given:
+
+        * collection_concept_id: Concept ID for collection that variables have
+          been generated for.
+        * cmr_env: URLs for CMR environments (OPS, UAT, and SIT)
+        * auth_header: Authorization HTTP header, containing either:
+          - A LaunchPad token: 'Authorization: <token>'
+          - An EDL bearer token: 'Authorization: Bearer <token>'
+        * publish: Optional argument determining whether to publish the
+          generated UMM-Var records to the indicated CMR instance. Defaults to
+          False.
+
+        Note - if attempting to publish to CMR, a LaunchPad token must be used.
```

Comment:
As discussed in the PR description - it turns out this isn't currently true. But it likely will be soon, so I didn't want to confuse matters for future users.